src/aatoolbox/datasources/glofas/glofas.py

```diff
@@ -0,0 +1,463 @@
+
+        Parameters
+        ----------
+        clobber : bool, default = False
+            Overwrite files that were already processed
+
+        Returns
+        -------
+        A list paths of processed files
+
+        """
+        logger.info(
+            f"Processing GloFAS {self._forecast_type} for "
+            f"{self._start_date} - {self._end_date} and up to "
+            f"{self._leadtime_max} day lead time"
+        )
+        # Make the directory
+        output_directory = self._get_directory(is_processed=True)
+        output_directory.mkdir(parents=True, exist_ok=True)
+        # Get list of files to open
+        processed_filepaths = []
+        for date in self._date_range:
+            input_filepath = self._get_filepath(
+                year=date.year,
+                month=date.month,
+                day=date.day,
+                leadtime_max=self._leadtime_max,
+            )
+            output_filepath = self._get_filepath(
+                year=date.year,
+                month=date.month,
+                day=date.day,
+                leadtime_max=self._leadtime_max,
+                is_processed=True,
+            )
+            if not clobber and output_filepath.exists():
+                continue
+            logger.debug(f"Processing {input_filepath}")
+            ds_raw = self._load_single_file(
+                input_filepath=input_filepath,
+                filepath=output_filepath,
+                clobber=clobber,
+            )
+            ds_processed = self._get_reporting_point_dataset(ds=ds_raw)
+            # NetCDF doesn't like to overwrite files
+            if output_filepath.exists():
+                output_filepath.unlink()
+            ds_processed.to_netcdf(output_filepath)
+            processed_filepaths.append(output_filepath)
+            logger.debug(f"Wrote file to {output_filepath}")
+        logger.info(
+            f"Processed {len(processed_filepaths)} files to {output_directory}"
+        )
+        logger.debug(f"Files downloaded: {processed_filepaths}")
+        return processed_filepaths
+
+    def load(
+        self,
+    ) -> xr.Dataset:
+        """
+        Load the processed GloFAS data as an xarray.DataSet.
+
+        Returns
+        -------
+        A single xarray dataset containing all GloFAS reporting points
+        and their associated river discharge
+
+        """
+        filepath_list = [
+            self._get_filepath(
+                year=date.year,
+                month=date.month,
+                day=date.day,
+                leadtime_max=self._leadtime_max,
+                is_processed=True,
+            )
+            for date in self._date_range
+        ]
+        with xr.open_mfdataset(
+            filepath_list, preprocess=self._preprocess_load
+        ) as ds:
+            return ds
+
+    def _get_filepath(
+        self,
+        year: int,
+        month: int = None,
+        day: int = None,
+        leadtime_max: int = None,
```

Comment:
This doesn't need to be an explicit argument I think, can just be referenced internally with `self._`

Reply:
Ah great catch, thanks!